ecs/pod_creator.go

```diff
@@ -27,11 +31,227 @@ func NewBasicECSPodCreator(c cocoa.ECSClient, v cocoa.Vault) (*BasicECSPodCreato
 
 // CreatePod creates a new pod backed by AWS ECS.
 func (m *BasicECSPodCreator) CreatePod(ctx context.Context, opts ...*cocoa.ECSPodCreationOptions) (cocoa.ECSPod, error) {
-	return nil, errors.New("TODO: implement")
+
+	mergedPodCreationOpts := cocoa.MergeECSPodCreationOptions(opts...)
+	mergedPodExecutionOpts := cocoa.MergeECSPodExecutionOptions(mergedPodCreationOpts.ExecutionOpts)
+
+	err := mergedPodCreationOpts.Validate()
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid pod creation options")
+	}
+
+	err = mergedPodExecutionOpts.Validate()
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid pod execution options")
+	}
+
+	allEnvVars, allSecrets, allSecretsToCreate, err := m.exportSecrets(ctx, mergedPodCreationOpts)
```

Comment:
This shouldn't return 4 values from a function, since returning more than 2 (or maybe 3) values is generally a sign that the code is trying to do too many things at once. It's preferable to make multiple passes over all the container definitions than try to squeeze all the information you can out of one pass through the slice because the number of container definitions is going to be small and it'll be less cluttered if you return only one thing per function call.